### PR TITLE
[MM-33713] Add validation for rotator when using rotator for cluster upgrades via API

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
- Add validation for rotator when using rotator for cluster upgrades via API

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33713

#### Release Note
```release-note
- Add validation for rotator when using rotator for cluster upgrades via API
```

### CODE REVIEW

**Review by stafot — APPROVED** [review]
LGTM. thanks!

File: model/cluster_request.go [hunk 1]
```diff
@@ -165,6 +165,30 @@ func (p *PatchUpgradeClusterRequest) Validate() error {
 		return errors.Errorf("unsupported cluster version %s", *p.Version)
 	}
 
+	if p.RotatorConfig != nil {
+		if p.RotatorConfig.UseRotator == nil {
+			return errors.Errorf("Rotator config use rotator should be set")
```
Comment (gabrieljackson) [inline model/cluster_request.go:170]: nit: lowecase `rotatator config use....` for this return error and the ones below.

**Review by gabrieljackson — APPROVED** [review]
Good fix! No blockers on my end.

It may make sense to set defaults for some of these values in the future. We do this for other API requests such as creating installations. Either way, good to make sure the validation is correct.

**Comment by stylianosrigas** [comment]
> Good fix! No blockers on my end.
> 
> It may make sense to set defaults for some of these values in the future. We do this for other API requests such as creating installations. Either way, good to make sure the validation is correct.

Yeap I was thinking about and was not sure if we want to enforce it. For now I just set these checks to avoid any upgrade fails.

**Pull request merged**